src/EditorComponent.jsx

```diff
@@ -163,6 +202,53 @@ const EditorComponent = props => {
             return true;
         };
 
+        // Editor command that simply tells the extension to capture the current frame. Does not apply any
+        // transaction to the state.
+        const tellExtensionToCaptureFrame = (state, dispatch) => {
+            const { videoId } = parentApp.currentVideoInfo();
+
+            if (!videoId) {
+                return false;
+            }
+
+            window.frames[0].postMessage({ type: AppConfig.CaptureCurrentFrameMessage }, '*');
+
+            return true;
+        };
+
+        // Listener that checks receiving the data url from extension.
+        const gotResponseFromExtension = e => {
+            if (!editorView.current) {
+                return;
+            }
+
+            if (e.data.type === AppConfig.CaptureCurrentFrameResponse) {
```

Comment:
Use negative condition to exist early. E.g

`if (e.data.type !== AppConfig.CaptureCurrentFrameResponse) return;`

and then rest of the function

Reply:
Rather, above condition can be combined with this. E.g.
`if (!editorView.current || e.data.type !== AppConfig.CaptureCurrentFrameResponse) return`